web/modules/custom/joinup_notification/joinup_notification.behat.inc

```diff
@@ -149,6 +149,114 @@ class JoinupNotificationSubContext extends DrupalSubContextBase implements Drupa
   }
 
   /**
+   * Asserts that a system email has been sent.
+   *
+   * Table format:
+   * | recipient      | username0123     |
+   * | recipient_mail | user@example.com |
+   * | subject        | The mail subject |
+   * | body           | The message body |
+   * | html           | yes/no           |
+   *
+   * The 'body' is a partial text based match.
+   * The 'html' can be 'yes' or 'no'. When not present, 'yes' is assumed.
+   *
+   * @param \Behat\Gherkin\Node\TableNode $table
+   *   The mail data table.
+   *
+   * @throws \Exception
+   *   Throws an exception when a parameter is not the expected one.
+   *
+   * @Then the following system email should have been sent:
+   */
+  public function assertSystemEmailSent(TableNode $table) {
+    /** @var string $body */
+    /** @var string $recipient */
+    /** @var string $recipient_mail */
+    /** @var string $subject */
+    extract($table->getRowsHash());
```

Comment:
There is also the `$html` variable being extracted. BTW this would be better with normal use of the rows so we can use human friendly column names without underscores.